spring-rabbit/src/main/java/org/springframework/amqp/rabbit/annotation/MultiRabbitListenerAnnotationBeanPostProcessor.java

```diff
@@ -70,14 +73,30 @@ private RabbitListener proxyIfAdminNotPresent(final RabbitListener rabbitListene
 	 * @return The name of the RabbitAdmin bean.
 	 */
 	protected String resolveMultiRabbitAdminName(RabbitListener rabbitListener) {
-		String admin = super.resolveExpressionAsString(rabbitListener.admin(), "admin");
-		if (!StringUtils.hasText(admin) && StringUtils.hasText(rabbitListener.containerFactory())) {
-			admin = rabbitListener.containerFactory() + RabbitListenerConfigUtils.MULTI_RABBIT_ADMIN_SUFFIX;
+
+		if (StringUtils.hasText(rabbitListener.admin())) {
+
+			var admin = super.resolveExpression(rabbitListener.admin());
+			if (admin instanceof RabbitAdmin rabbitAdmin) {
+
+				return rabbitAdmin.getBeanName();
+			}
+
+			return super.resolveExpressionAsString(rabbitListener.admin(), "admin");
```

Comment:
I don't think we need to call `(rabbitListener.admin()` so may times.
Why cannot we just extract a local variable for its result?